Build errors sent to stdout are not shown when Carthage fails

This isn't really our bug, but it's really unhelpful when `xcodebuild` fails and all the Carthage outputs is something like this (from #215):

```
Error Domain=org.carthage.ReactiveTask Code=0 "** BUILD FAILED **


The following build commands failed:
    Check dependencies
(1 failure)
" UserInfo=0x7fdb6a6394c0 {NSLocalizedDescription=** BUILD FAILED **


The following build commands failed:
    Check dependencies
(1 failure)
, ReactiveTaskErrorExitCode=65, ReactiveTaskErrorStandardError=** BUILD FAILED **


The following build commands failed:
    Check dependencies
(1 failure)
}
```

I've written a CI script before that simply finds all lines starting with something like `(failed|error|warning):` and aggregates those upon a failure. A similar approach might be useful here.

--- a/Source/carthage/Build.swift
+++ b/Source/carthage/Build.swift
@@ -11,6 +11,7 @@ import Commandant
 import Foundation
 import LlamaKit
 import ReactiveCocoa
+import ReactiveTask
 
 public struct BuildCommand: CommandType {
 	public let verb = "build"
@@ -30,6 +31,18 @@ public struct BuildCommand: CommandType {
 				let directoryURL = NSURL.fileURLWithPath(options.directoryPath, isDirectory: true)!
 
 				let (stdoutSignal, schemeSignals) = self.buildProjectInDirectoryURL(directoryURL, options: options)
+
+				// Redirect any error-looking messages from stdout, because
+				// Xcode doesn't always forward them.
+				var grepDisposable: Disposable?
+				if !options.verbose {
+					let coldOutput = stdoutSignal.replay(0)
+					let task = TaskDescription(launchPath: "/usr/bin/grep", arguments: [ "--extended-regexp", "(warning|error|failed):" ], standardInput: coldOutput)
+
+					let stderrSink: FileSink<NSData> = FileSink.standardErrorSink()
+					grepDisposable = launchTask(task, standardOutput: SinkOf(stderrSink)).start()
+				}
+
 				stdoutSignal.observe(stdoutSink)
 
 				let formatting = options.colorOptions.formatting
@@ -42,6 +55,9 @@ public struct BuildCommand: CommandType {
 						}
 					}, next: { (project, scheme) in
 						carthage.println(formatting.bullets + "Building scheme " + formatting.quote(scheme) + " in " + formatting.projectName(string: project.description))
+					}, disposed: {
+						grepDisposable?.dispose()
+						return
 					})
 					.then(.empty())
 			}

